Automattic/sensei
# Add/e2e students management

### Changes proposed in this Pull Request
* It configures the first e2e test using playwriting 
* It uses the https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/ to create the WordPress env.
* it is based on https://github.com/WordPress/gutenberg/tree/trunk/packages/e2e-test-utils-playwright and https://github.com/WordPress/gutenberg/tree/trunk/test/e2e
* It is covering only the `add to course` action on the student `action menu`. 
* It is adding Page Object Modal to organize page selectors.

### Extra information
* We can use the amazing playwright VSCode plugin https://marketplace.visualstudio.com/items?itemName=ms-playwright.playwright to help us write and debug the e2e tests.
* I have some issues running wpcli tasks inside the `@wordpress/env`  docker images, so for now we are not using `wpcli` tasks to create `Course` or a `Student`, but this approach is not scalable. It requires further investigation. The issue is related to running commands with flags `command ---flag-name` or params `command param`, any flag or param is reaching the PHP command file.
* It is not using the package https://github.com/WordPress/gutenberg/tree/trunk/packages/e2e-test-utils-playwright , because the package is still experimental and focused on the Gutenberg repo requirement. Currently, we already have a lot of other libraries on our codebase that are pretty hard to maintain or update, so we need to be careful to add new packages.

### Tasks:

- [x] - Setup playwright locally 
- [x] - Setup playwright on CI 
- [x] - Reset database before run the test suite
- [x] - Enable VSCode to run
- [ ] - Update e2e test to cover the last `Enrolled Courses` changes
- [ ] -  Use `WP Rest api` or `wpcli` to create the data scenarios (Other PR)
- [ ] - Convert the existent e2e to use playwright (Other PR).
- [ ] - Pipeline sometimes is not able to start/clean up the wp-env properly 


### Testing instructions
* Run `npm install`.
* Run `npx playwright install`
* Run `npm run test:e2e-playwright` or `npm run test:e2e-playwright :debug`  if you want to see the browser.
